src/components/menu/examples/menu-open-sub-menu-programatically.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+    MenuItem,
+    ListSeparator,
+    LimelMenuCustomEvent,
+} from '@limetech/lime-elements';
+import { Component, State, h, Host } from '@stencil/core';
+import { CascadingMenuItems } from './item-constants';
+
+/**
+ * Opening sub-menus programatically
+ *
+ * It is possible to open any sub-menu in the menu-hierarchy.
+ * This is done by using the parentItem property of the MenuItem class.
+ * @link item-constants.ts
+ */
+@Component({
+    tag: 'limel-example-menu-open-sub-menu-programatically',
+    shadow: true,
+    styleUrl: 'menu-open-sub-menu-programatically.scss',
+})
+export class MenuOpenSubMenuProgramaticallyExample {
+    private readonly rootItems: Array<MenuItem | ListSeparator> =
+        CascadingMenuItems;
+
+    @State()
+    private items: Array<MenuItem | ListSeparator> = this.rootItems;
+
+    @State()
+    private lastSelectedItem: string;
+
+    @State()
+    private selectedMenuItem: MenuItem;
+
+    @State()
+    private openMenu: boolean = false;
+
+    public render() {
+        return (
+            <Host>
+                {this.renderMenu()}
+                <limel-example-value
+                    label="Last selected item"
+                    value={this.lastSelectedItem}
+                />
+            </Host>
+        );
+    }
+
+    private renderMenu() {
+        return (
+            <div class="menu-container">
+                <limel-menu
+                    items={this.items}
+                    open={this.openMenu}
+                    selectedMenuItem={this.selectedMenuItem}
+                    onSelect={this.handleSelect}
+                    onNavigateMenu={async (
+                        event: LimelMenuCustomEvent<MenuItem>
+                    ) => {
+                        if (!event.detail) {
+                            this.items = this.rootItems;
+                        }
+                    }}
+                    onCancel={() => {
+                        this.items = this.rootItems;
+                        this.openMenu = false;
+                    }}
+                >
+                    <limel-button label="Menu" slot="trigger" />
+                </limel-menu>
+                <limel-button
+                    label='Shortcut to "Bullets and numbering"'
+                    primary={true}
+                    onClick={this.buttonClick}
+                />
+            </div>
+        );
+    }
+
+    private buttonClick = () => {
```

Comment:
I'm moving this comment since the thread it was in was resolved. Since the code has been updated, it's not describing the current state of the code 100 % correctly anymore, but the gist of it is still relevant:

---

Hmm… hang on… in the `CascadingMenuItems` imported from `item-constants.ts`, the sub-menu items actually do _contain_ the items included in that sub-menu. Why is it different here?

Another difference is that all the items in `selectedItems` have a reference to their `parentItem`, which apparently isn't necessary when not opening the menu programmatically?

The "vanilla" example for cascading menu is so clean, you just supply a nested structure of items, and everything is just taken care of, but here, there is so much extra stuff to do, you have to create new items yourself, and replace the normal structure of items with this specially crafted one, and then remember to restore the normal structure again at the right moments (plural!).

Have you explored the possibility of keeping the simple "rootItems" and just telling the menu where to navigate to, somehow? I imagine you could provide it with a "path". I guess since it's all arrays, and `value` is optional, rather than a nice `'format.lists'`, you'd give it something like `'0.2'` or `'[0][2]'`, but still, it would be much less work on the consumer.

By the way, I just noticed this little curiosity, which I think highlights why it would be nice to not have the consumer do all this "manual labour":

https://github.com/Lundalogik/lime-elements/assets/154725/e698622a-73bf-4d0e-abed-f3ca118aab6d

Since I feel this feedback might be a little hard to hear, considering it might require some rework, I wan't to express that the cascading menu feature in itself is absolutely fantastic, and you've really done some great work here!

Is the ability to programmatically open the menu to any sub-menu critical for your use-case? If not, I would suggest simply leaving that functionality out for now, and add the feature once it's ready and stable.

Reply:
Yeah, I know that it isn't a very good design on this specific feature, YES it is critical in our implementation of the hierarchy picker. But maybe we could solve it from the outside some how. The reason the "path thing" doesn't work is because when we later enable lazy-loading which we already have in the next PR, the items isn't arrays anymore but functions that load the data when clicked. Because the limel-menu doesn't have all the data of all hierarchies it will be hard to define. We've tried to simplify it but haven't find a good way.

I have updated the example so it fetches the items from the rootItems instead of defining them again, that might make the example a little bit easier to read. If you have any ideas on how to make it easier I'm all ears, or if you want to talk and discuss it.

Reply:
Ah, I understand. Yeah that seems like it would make the path approach more difficult.

Then I would suggest that we leave the functionality in, but actually remove the example, and add something to the docs about `selectedMenuItem` being an experimental property which shouldn't be used, and which we may remove without warning in the future. Then _you_ can use it, but we don't promise anyone else we'll keep it, if we find a better solution. And if we don't, we can make it permanent and add the example back at a later time.

Sound ok?

Reply:
Yeah, sounds like a plan.
Then we can make the `parentItem` property on `MenuItem` private as well.

Reply:
Done, make sure to verify that you agree with the changes, and if so, Resolve the conversation :)